src/container/srv_target.c

```diff
@@ -1306,18 +1328,24 @@ cont_close_hdl(uuid_t cont_hdl_uuid)
 		return 0;
 	}
 
-	D_ASSERT(hdl->sch_cont != NULL);
+	cont_child = hdl->sch_cont;
+	D_ASSERT(cont_child != NULL);
 
-	D_DEBUG(DF_DSMS, DF_CONT": closing (%s): hdl="DF_UUID"\n",
-		DP_CONT(hdl->sch_cont->sc_pool->spc_uuid,
-			hdl->sch_cont->sc_uuid),
-		hdl->sch_cont->sc_closing ? "resent" : "new",
-		DP_UUID(cont_hdl_uuid));
+	D_DEBUG(DF_DSMS, DF_CONT": closing (%d): hdl="DF_UUID"\n",
+		DP_CONT(cont_child->sc_pool->spc_uuid,
+			cont_child->sc_uuid),
+		cont_child->sc_open, DP_UUID(cont_hdl_uuid));
+
+	/* Remove the handle from hash first, following steps may yield */
+	ds_cont_local_close(cont_hdl_uuid);
```

Comment:
why not call cont_hdl_delete() directly since you have the @hdl ?

Reply:
yes, we can just call cont_hdl_delete() directly, let's cleanup it in later patches.